src/ahp/prover.rs

```diff
@@ -456,21 +484,30 @@ impl<F: PrimeField> AHPForR1CS<F> {
             r_alpha_evals.evals.clone().into_iter().step_by(step).collect(),
             domain_h.clone()
         );
-        let mut z_1 = BoundaryPolynomial::from_coboundary_polynomial_evals(
-            outer_poly_evals_on_H
-        ).unwrap().polynomial();
-
-        // Let b = zk_bound. This blinding polynomial is of type
-        // (c_0 + ... c_(2b-1) * X ^ (2b - 1) )
-        // since the boundary poly is queried twice as much as the others
-        let mut randomization_poly = Polynomial::rand(2 * zk_bound - 1, rng);
-        randomization_poly = randomization_poly.mul_by_vanishing_poly(domain_h.size());
-
-        // Add the blinding polynomial to z_1
-        z_1 = &z_1 + &randomization_poly;
-        end_timer!(z_1_time);
+        let (z_1, rand_poly_degree) = {
+            let z_1_t = BoundaryPolynomial::from_coboundary_polynomial_evals(
+                outer_poly_evals_on_H
+            ).unwrap().polynomial();
+
+            if zk {
+                // Let b = zk_bound. This blinding polynomial is of type
+                // (c_0 + ... c_(2b-1) * X ^ (2b - 1) )
+                // since the boundary poly is queried twice as much as the others
+                let mut randomization_poly = Polynomial::rand(2 * zk_bound - 1, rng);
+                randomization_poly = randomization_poly.mul_by_vanishing_poly(domain_h.size());
```

Comment:
The comment is a bit misleading, because it only holds for our choice of zk_bound = 2. 
Let us simply write: The boundary polynomial is queried three times overall (the double number of 
ordinary queries, plus the one extra in the multi-point open), hence we need four coefficients of randomness.